src/python/generate_stubs.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import absolute_import, annotations, division, print_function
+
+import mypy.stubgen
+import mypy.stubgenc
+from mypy.stubgenc import SignatureGenerator, DocstringSignatureGenerator
+
+from stubgenlib.siggen import (
+    AdvancedSignatureGenerator,
+    AdvancedSigMatcher,
+)
+from stubgenlib.utils import add_positional_only_args
+
+
+PY_TO_STDVECTOR_RESULT = "float | list[float] | tuple[float, ...]"
+
+
+class OIIOSignatureGenerator(AdvancedSignatureGenerator):
+    sig_matcher = AdvancedSigMatcher(
+        signature_overrides={
+            # signatures for these special methods include many inaccurate overloads
+            "*.__ne__": "(self, other: object) -> bool",
+            "*.__eq__": "(self, other: object) -> bool",
+        },
+        arg_type_overrides={
+            # FIXME: Buffer may in fact be more accurate here
+            ("*", "*", "Buffer"): "numpy.ndarray",
+            # these use py_to_stdvector util
+            ("*.ImageBufAlgo.*", "min", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "max", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "black", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "white", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "sthresh", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "scontrast", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "white_balance", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "values", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "top", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "bottom", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "topleft", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "topright", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "bottomleft", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "bottomright", "object"): PY_TO_STDVECTOR_RESULT,
+            ("*.ImageBufAlgo.*", "color", "object"): PY_TO_STDVECTOR_RESULT,
+            # BASETYPE & str are implicitly converible to TypeDesc
+            ("*", "*", "*.TypeDesc"): "Union[TypeDesc, BASETYPE, str]",
+            # list is not strictly required
+            ("*.ImageOutput.open", "specs", "list[ImageSpec]"): "Iterable[ImageSpec]",
+        },
+        result_type_overrides={
+            # FIXME: is there a way to use std::optional for these?
+            ("*.ImageOutput.create", "object"): "ImageOutput | None",
```

Comment:
I tried to modify the wrapper code to use `std::optional`, but it complained about the classes being abstract.